x-pack/elastic-agent/pkg/agent/application/paths/paths.go

```diff
@@ -37,6 +43,16 @@ func Top() string {
 	return topPath
 }
 
+// TempDir returns agent temp dir located within data dir.
+func TempDir() string {
+	tmpDir := filepath.Join(Data(), tempSubdir)
+	tmpCreator.Do(func() {
```

Comment:
Not sure in which case this is executed multiple times? Tests?

Reply:
tempDir is called on each install, i want to avoid Mkdir to be executed multiple times

Reply:
The two things I'm trying to understand:

* What is the problem of executing it multiple times
* Do we have a case where this might happen

My goal is to understand what the driver for this is. I don't see any issue with the code itself.

Reply:
Actually I think it would be better to call `MkdirAll` every time. If it already exists it will do nothing, but it also ensures that before going further in this code path it does exist.

The user might have gone in and deleted or something while Agent is already running, so its better to just make sure its there before going any further.

Reply:
@michalpristas you might have to follow up on that as you just merged ^